src/serializer/ResidualHeapSerializer.js

```diff
@@ -1280,10 +1280,12 @@ export class ResidualHeapSerializer {
         let value = this.realm.getGlobalLetBinding(boundName);
         // Check for let binding vs global property
         if (value) {
-          let id = this.serializeValue(value, true, "let");
+          let rval = residualFunctionBinding.value;
+          invariant(rval !== undefined && value.equals(rval));
```

Comment:
Hm, this shows that the way CSE is currently done in the visitor is risky.
There's an invariant that the set (at least at the cardinality level) of "serialized" values should be the same as the set of "visited" values. I wonder why that didn't fire here, as it seems like there should have been a mismatch?

Reply:
Good question. I don't know.

Reply:
The reason just came to me: The invariant would have failed, had not another invariant failed first. This changes makes both invariant checks pass.